src/server/scripts/Kalimdor/zone_silithus.cpp

```diff
@@ -532,21 +532,26 @@ class npc_anachronos_the_ancient : public CreatureScript
                         break;
                     case 51:
                         {
-                            uint32 entries[4] = { 15423, 15424, 15414, 15422 };
-                            for (uint8 i = 0; i < 4; ++i)
+                            std::list<Creature*> constructList;
+
+                            me->GetCreatureListWithEntryInGrid(constructList, 15423, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15424, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15414, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15422, 100.0f);
```

Comment:
Every call of `GetCreatureListWithEntryInGrid` overrides the previous container, so only mobs from the last call are removed from world.